src/Core/Grid/Action/Row/AccessibilityChecker/DeleteCartAccessibilityChecker.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Action\Row\AccessibilityChecker;
+
+/**
+ * Checks if delete option for cart can be performed.
+ */
+final class DeleteCartAccessibilityChecker implements AccessibilityCheckerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function isGranted(array $record)
+    {
+        // if its number then the order has been placed so the cart cant be removed.
+        return !is_numeric($record['status']);
```

Comment:
```suggestion
        return isset($record['status']) && is_int($record['status']) && 0 => $record['status'];
```

Reply:
Does  `&& 0 => $record['status'];` should be `&& 0 == $record['status'];` ?

Reply:
` return isset($record['status']) && !is_int($record['status']) && 0 == $record['status'];`

Reply:
Regarding the comment, you're checking the order has an integer status.
With `is_numeric` you're allowing `9.2, `02471`, `-100`.
And because the method is public, you need to control that the array has `status` available :)
Of course unit tests are helpful here, but not necessary but if you have time to do it, could be awesome :+1:

Reply:
added the unit test. Without it there is no proof that I understand this code. I like the tests. Without them feel like always taking a loan from the bank which one day will have to pay. Thank for test supportive attitude :tada:

Reply:
> added the unit test. Without it there is no proof that I understand this code. I like the tests. Without them feel like always taking a loan from the bank which one day will have to pay. Thank for test supportive attitude 🎉

Nice !